src/FloatingButton/FloatingButton.tsx

```diff
@@ -0,0 +1,199 @@
+import * as classnames from "classnames";
+import * as PropTypes from "prop-types";
+import * as React from "react";
+
+import { Button } from "../Button/Button";
+import FlexBox from "../flex/FlexBox";
+
+import "./FloatingButton.less";
+
+const cssClass = {
+  ACTIVE: "FloatingButton--active",
+  inactive: positionY => `FloatingButton--inactive--${positionY}`,
+
+  propStyle: prop => `FloatingButton--${prop}`,
+
+  BUTTON: "FloatingButton--button",
+  CONTAINER: "FloatingButton--container",
+  REMOVE: "FloatingButton--remove",
+  SUB_ELEMENT: "FloatingButton--subElement",
+};
+
+const propTypes = {
+  animate: PropTypes.bool,
+
+  additionalButtons: PropTypes.array,
+
+  bgColor: PropTypes.string,
+  className: PropTypes.string,
+  closeLabel: PropTypes.node,
+  label: PropTypes.node,
+
+  onClick: PropTypes.func,
+
+  offsetX: PropTypes.number,
+  offsetY: PropTypes.number,
+  positionX: PropTypes.oneOf(["left", "center", "right"]),
+  positionY: PropTypes.oneOf(["bottom", "top"]),
+  size: PropTypes.string,
+};
+
+const defaultProps = {
+  animate: true,
+  label: "+",
+  offsetX: 30,
+  offsetY: 30,
+  positionX: "right",
+  positionY: "bottom",
+};
+
+export default class FloatingButton extends React.PureComponent {
+  static propTypes = propTypes;
+  static defaultProps = defaultProps;
+  static cssClass = cssClass;
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      active: false,
+      display: false,
+    };
+  }
+
+  horizontalStyle() {
+    const { offsetX, positionX } = this.props;
+    if (positionX === "center") return {};
+    return positionX === "left"
+      ? {
+          left: `${offsetX}px`,
+        }
+      : {
+          // default to right if a bad positionX input provided
+          right: `${offsetX}px`,
+        };
+  }
+
+  verticalStyle() {
+    const { offsetY, positionY } = this.props;
+    return positionY === "top"
+      ? {
+          top: `${offsetY}px`,
+        }
+      : {
+          // default to bottom orientation
+          bottom: `${offsetY}px`,
+        };
+  }
+
+  mainButtonHandler = () => {
+    const { additionalButtons, animate, onClick } = this.props;
+    const { active } = this.state;
+    if (additionalButtons && additionalButtons.length) {
+      // main button activates additional buttons
+      if (active) {
+        // stop displaying when transition to inactive state complete
+        this.setState({ active: false, display: true });
+        setTimeout(() => this.setState({ display: false }), animate ? 200 : 0);
+      } else {
+        this.setState({ active: true, display: true });
+      }
+    }
+    if (onClick && !active) {
```

Comment:
The name `onClick` implies that this event will fire every time our component is clicked. If this function only fires to open the menu, can we rename it to `onOpen` or something that more clearly illustrates that it will not actually fire on every click?

Reply:
I don't want to call it onOpen because when there's no button menu to open, this fires on every click (since we never enter the active state). Happy to remove this logic so it fires on every click though.

Reply:
I removed the active check